### DISCUSSION TITLE
How to use a own qrel file?

### DISCUSSION THREAD

**Discussion by phsm99** [castorini/pyserini#1549] [Q&A]
I have a file that have the QueryId,DocID,Relevance and i want to use it to train the model. Can i do it?

**Comment by octocat20** [comment]
For evaluation, yes. Qrels are relevance judgments, not a training hook.

Put them in TREC format (4 columns):

qid 0 docid relevance

Example:
1 0 D001 1
1 0 D002 0
2 0 D010 2

Then:

python -m pyserini.eval.trec_eval -c -m map -m ndcg_cut.10 /path/to/your.qrels /path/to/run.txt

If you have CSV (QueryId,DocID,Relevance), convert to that 4-column form first (insert a dummy 0 as the second field).

get_qrels("msmarco-passage") is only for built-in sets. For your file, pass the path straight into trec_eval. That scores a run. It does not train anything by itself.
